core/controllers/acl_decorators.py

```diff
@@ -2753,6 +2753,82 @@ def test_can_accept_suggestion(
     return generate_decorator_for_handler
 
 
+def can_reviewer_or_suggester_update_suggestions(handler):
+    """Decorator to check whether user can update the suggestions.
+
+    Args:
+        handler: function. The function to be decorated.
+
+    Returns:
+        function. The newly decorated function that has common checks and
+        permissions specified by passed in decorator. This function is allowed
+        to be made by the users who have the rights to accept suggestions and
+        the suggesters who made their own suggestions.
+
+    Raises:
+        NotLoggedInException. The user is not logged in.
+        UnauthorizedUserException. The user does not have credentials to
+            edit this suggestion.
+        InvalidInputException. The submitted suggestion id is not valid.
+        PageNotFoundException. A suggestion is not found with the given
+            suggestion id.
+    """
+    def test_can_update_suggestion(
+            self, suggestion_id, **kwargs):
+        """Returns a handler to test whether a
+        suggestion can be updated based on the user's roles.
+
+        Args:
+            suggestion_id: str. The suggestion id.
+            **kwargs: *. Keyword arguments.
+
+        Returns:
+            function. The handler for updating a
+            suggestion.
+
+        Raises:
+            NotLoggedInException. The user is not logged in.
+            UnauthorizedUserException. The user does not have credentials to
+                edit this suggestion.
+            InvalidInputException. The submitted suggestion id is not valid.
+            PageNotFoundException. A suggestion is not found with the given
+                suggestion id.
+        """
+        if not self.user_id:
+            raise base.UserFacingExceptions.NotLoggedInException
+        user_actions = self.user.actions
+
+        if len(suggestion_id.split('.')) != 3:
+            raise self.InvalidInputException(
+                'Invalid format for suggestion_id.'
+                ' It must contain 3 parts separated by \'.\'')
+
+        suggestion = suggestion_services.get_suggestion_by_id(suggestion_id)
+
+        if suggestion is None:
+            raise self.PageNotFoundException
+
+        if suggestion.suggestion_type == (
+                feconf.SUGGESTION_TYPE_TRANSLATE_CONTENT):
+            if user_services.can_review_translation_suggestions(
+                    self.user_id,
+                    language_code=suggestion.change.language_code):
+                return handler(self, suggestion_id, **kwargs)
+
+        if role_services.ACTION_ACCEPT_ANY_SUGGESTION in user_actions:
+            return handler(self, suggestion_id, **kwargs)
```

Comment:
Assume `suggestion.suggestion_type` is question, in that case do we want to run the handler?

Reply:
Once this feature is implemented for updating question suggestions, another check will be added here to check necessary authorities to be have by the user to update a queston suggestion.

Reply:
Shouldn't we do that in a follow-up PR, then? Validating non-translation suggestions seems to be outside the scope of this change (considering the context of the PR & the tests that were added for the new decorator).

Reply:
Yes. I hope to use the same decorator for question suggestion requests as well. It will be a part of this issue https://github.com/oppia/oppia/issues/11735

Reply:
In that case, I think we shouldn't be checking for permissions beyond the ones needed for translations in this PR. The next PR could include the bits needed for questions, but we shouldn't implement functionality that's not going to be used yet.

Given that, is the only check we need ``can_review_translation_suggestions``?

Reply:
I have a different perspective than Ben's and probably one that is closer to Sandeep's. I would suggest raising an exception if the suggestion type is not "translation".

The line that this comment is commenting on seems fine to me from a permissions perspective (anyone who can accept suggestions should be able to edit them before accepting). I believe this is reserved only for admins.